test/solarwinds_apm/config_test.rb

```diff
@@ -0,0 +1,274 @@
+
+    it 'with wrong env, use default true ' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with wrong env, use default true ' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with config env true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with config env false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, false, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, config env true, use default false, should be true' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, empty config env, use default false, should be false' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, '', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with nil env, empty config env, use default true, should be true' do
+      SolarWindsAPM::Config.enable_disable_config(nil, :dummy_key, '', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+  end
+
+  describe '#set_log_level' do
+    before do
+      SolarWindsAPM::Config.initialize
+      @current_debug_level = ENV['SW_APM_DEBUG_LEVEL']
+      ENV.delete('SW_APM_DEBUG_LEVEL')
+    end
+
+    after do
+      ENV['SW_APM_DEBUG_LEVEL'] = @current_debug_level
+    end
+    
+    it 'debug_level is out of range use default 1' do
+      SolarWindsAPM::Config[:debug_level] = 7
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 1
+    end
+
+    it 'debug_level is in the range' do
+      SolarWindsAPM::Config[:debug_level] = 1
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 3
+    end
+
+    it 'env var override config' do
+      ENV['SW_APM_DEBUG_LEVEL'] = '3'
+      SolarWindsAPM::Config[:debug_level] = 1
+      SolarWindsAPM::Config.set_log_level
+      _(SolarWindsAPM.logger.level).must_equal 1
```

Comment:
The test says env var should override but the result looks the opposite, why?

Reply:
When SW_APM_DEBUG_LEVEL = 3, the mapping will result the SolarWindsAPM.logger.level to 1.
If SolarWindsAPM::Config[:debug_level] = 1 (and empty `SW_APM_DEBUG_LEVEL`), then the mapping will result the SolarWindsAPM.logger.level to 3.